cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -93,9 +175,44 @@ func newDevicePlugin(sysfsDir, devfsDir string, options cliOptions) *devicePlugi
 		}
 	}
 
+	switch options.preferredAllocationPolicy {
+	case "balanced":
+		dp.policy = balancingPolicy
+	case "packed":
+		dp.policy = packingPolicy
+	default:
+		dp.policy = nonePolicy
```

Comment:
This approach can cause unexpected and hard to find issues when user mistypes command line option, e.g. ```-allocation-policy Balanced```. I'd suggest to improve validation of the user input and make this option case-insensitive.

Reply:
Right, but take the consideration of definition in operaters, I think it is better to restrict the valid value of `-allocation-policy` to only 3 `balanced, packed, none`.

Reply:
yes, but with current approach this option is not restricted to only 3 values. User can pass pretty much everything I believe.